src/types.lisp

```diff
@@ -590,15 +597,28 @@ The foreign array must be freed with foreign-array-free."
 (defmethod (setf foreign-struct-slot-value)
     (value ptr (slot aggregate-struct-slot))
   "Set the value of an aggregate SLOT to VALUE in PTR."
-  (translate-aggregate-to-foreign (inc-pointer ptr (slot-offset slot))
-                                  value
-                                  (parse-type (slot-type slot))))
+  (let ((type (slot-type slot))
+        (output-ptr (inc-pointer ptr (slot-offset slot))))
+    (if (= 1 (slot-count slot))
+        (translate-aggregate-to-foreign output-ptr value (parse-type type))
+        (loop for i below (slot-count slot)
+              for v in value
+              do (setf (mem-aref output-ptr type i)
+                       (translate-to-foreign v type))))))
```

Comment:
I think this should use `convert-into-foreign-memory` instead, but see my previous about piggybacking on the `:array` type.